Modules/Scripted/Endoscopy/Endoscopy.py

```diff
@@ -465,6 +523,42 @@ def step(self, segment, t, dl):
             pguess = p1
         return (t1, pguess, remainder)
 
+    def interpolateCameraTransforms(self):
+        # TODO: This approach works for the vtkMRMLMarkupsFiducialNode case.  We need to handle the
+        # vtkMRMLMarkupsCurveNode / vtkMRMLMarkupsClosedCurveNode case too, which is more difficult because it doesn't
+        # (yet) provide us self.pathIndices information.
```

Comment:
Since the `vtkMRMLMarkupsCurveNode`[^1] is based of `vtkMRMLMarkupsNode`[^2], it provides a convenient API to retrieve markup based of their index. See method named after the pattern `SetNthControlPoint/GetNthControlPoint`. The following functions should also be useful:
* `SetNthControlPointOrientationMatrix/GetNthControlPointOrientationMatrix`
* `SetNthControlPointOrientationMatrixWorld/GetNthControlPointOrientationMatrixWorld`

There is also an API
* 

[^1: https://github.com/Slicer/Slicer/blob/main/Modules/Loadable/Markups/MRML/vtkMRMLMarkupsCurveNode.h
[^2]: https://github.com/Slicer/Slicer/blob/main/Modules/Loadable/Markups/MRML/vtkMRMLMarkupsNode.h

Reply:
Thank you.  I am using this interface now.